src/driver/xdma_sgdma.c

```diff
@@ -261,22 +149,59 @@ ssize_t xdma_channel_read_write(struct xdma_channel *chnl,
 		return -EINVAL;
 	}
 
-	rv = check_transfer_align(engine, buf, count, *pos, 1);
-	if (rv) {
-		pr_info("Invalid transfer alignment detected\n");
-		return rv;
-	}
+	if (sg_alloc_table(&sgt, SG_MAX_SINGLE_ALLOC, GFP_KERNEL))
```

Comment:
Nit: I don't usually like to put functions into if statements unless they have an interrogative name (is/should/etc).

Imperative functions should be on their own line. So assign to a variable and then check.